great_expectations/profile/json_schema_profiler.py

```diff
@@ -0,0 +1,253 @@
+import abc
+from enum import Enum
+from typing import Any, Dict, Optional, Union
+
+from great_expectations.core import (
+    ExpectationConfiguration,
+    ExpectationKwargs,
+    ExpectationSuite,
+)
+
+
+class JsonSchemaTypes(Enum):
+    STRING = "string"
+    INTEGER = "integer"
+    NUMBER = "number"
+    ARRAY = "array"
+    NULL = "null"
+    BOOLEAN = "boolean"
+    OBJECT = "object"
+    ENUM = "enum"
+
+
+class NonDataProfiler(object, metaclass=abc.ABCMeta):
+    """
+    A base profiler that creates suites from non-data sources of truth.
+
+    Implementation notes:
+    - . _profile() is required
+    - .validate() is optional if additional validation is desired. Be sure to
+        call super().validate() before the child class custom logic.
+    """
+
+    def __init__(self, configuration: dict = None):
+        self.configuration = configuration
+
+    def validate(self, schema: dict) -> bool:
+        return isinstance(schema, dict)
+
+    def profile(self, schema: dict, suite_name: str) -> ExpectationSuite:
+        if not self.validate(schema):
+            raise TypeError(
+                f"The schema must be a dictionary and was given a {type(schema)}"
+            )
+        expectation_suite = self._profile(schema, suite_name=suite_name)
+        return expectation_suite
+
+    @abc.abstractmethod
+    def _profile(self, schema: dict, suite_name: str) -> ExpectationSuite:
+        pass
+
+
+class JsonSchemaProfiler(NonDataProfiler):
+    """
+    This profiler creates Expectation Suites from JSONSchema artifacts.
+
+    JSON Schema is a vocabulary that allows you to annotate and validate JSON
+    documents. https://json-schema.org
+
+    Basic suites can be created from these spedifications.
+
+    Note that there is not yet a notion of nested data types in Great
+    Expectations so suites generated use column map expectations.
+    """
+
+    def validate(self, schema: dict) -> bool:
+        result = super().validate(schema)
+        if not result:
+            return False
+
+        # TODO validate that the dict is a valid JSONSchema document. Pretty meta huh!
```

Comment:
Flagging TODO. `validate_schema` is available in json_schema, which is already a dependency for the core GE.

Reply:
I could not find an easy way to validate the schema files within my timebox. Per the spec literally any valid JSON can be a schema, however the spec does say "always use the `$schema`" key so I'm calling that an MVP.

Reply:
I dropped in a code snippet that will validate the schema using `jsonschema` --> it'll require adding that as an import at the top.